ra/ra.go

```diff
@@ -945,6 +956,19 @@ func (ra *RegistrationAuthorityImpl) validateFinalizeRequest(
 		return nil, berrors.BadCSRError("unable to parse CSR: %s", err.Error())
 	}
 
+	if ra.mustStapleAllowList != nil {
+		if issuance.ContainsMustStaple(csr.Extensions) {
```

Comment:
nit: I'd collapse these onto a single line, to prevent too much nesting, and to prevent readers from being confused about which of these three consecutive if clauses the `else` below corresponds to
```suggestion
	if ra.mustStapleAllowList != nil && issuance.ContainsMustStaple(csr.Extensions) {
```